encoding/value/timezone.rs

```diff
@@ -0,0 +1,731 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use std::fmt;
+
+use chrono::{FixedOffset, MappedLocalTime, NaiveDate, NaiveDateTime};
+use chrono_tz::Tz;
+
+use super::primitive_encoding::{decode_i32, decode_u32, encode_i32, encode_u32};
+
+#[derive(Copy, Clone, Debug)]
+pub enum Offset {
+    IANA(<Tz as chrono::TimeZone>::Offset),
+    Fixed(FixedOffset),
+}
+
+impl chrono::Offset for Offset {
+    fn fix(&self) -> FixedOffset {
+        match self {
+            Self::IANA(inner) => inner.fix(),
+            Self::Fixed(inner) => inner.fix(),
+        }
+    }
+}
+
+impl fmt::Display for Offset {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::IANA(inner) => fmt::Display::fmt(inner, f),
+            Self::Fixed(inner) => fmt::Display::fmt(inner, f),
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug)]
+pub enum TimeZone {
+    IANA(Tz),
+    Fixed(FixedOffset),
+}
+
+impl Default for TimeZone {
+    fn default() -> Self {
+        Self::IANA(Tz::default())
+    }
+}
+
+impl TimeZone {
+    pub(crate) fn encode(self) -> [u8; 4] {
+        match self {
+            TimeZone::IANA(tz) => encode_u32(encode_tz(tz)),
+            TimeZone::Fixed(fixed) => {
+                // from UTC-23:59:59 to UTC+23:59:59 <=> -86399..=86399
+                encode_i32(fixed.local_minus_utc())
+            }
```

Comment:
The named timezones are given values in `0..NUM_TZS`, while fixed offsets are in the range of `0x8000_0000_0000_000 ± 86399`.